src/python/gudhi/tensorflow/cubical_layer.py

```diff
@@ -0,0 +1,80 @@
+import numpy               as np
+import tensorflow          as tf
+from ..cubical_complex  import CubicalComplex
+
+######################
+# Cubical filtration #
+######################
+
+# The parameters of the model are the pixel values.
+
+def _Cubical(Xflat, Xdim, dimensions):
+    # Parameters: Xflat (flattened image),
+    #             Xdim (shape of non-flattened image)
+    #             dimensions (homology dimensions)
+
+    # Compute the persistence pairs with Gudhi
+    # We reverse the dimensions because CubicalComplex uses Fortran ordering
+    cc = CubicalComplex(dimensions=Xdim[::-1], top_dimensional_cells=Xflat)
+    cc.compute_persistence()
+
+    # Retrieve and ouput image indices/pixels corresponding to positive and negative simplices    
+    cof_pp = cc.cofaces_of_persistence_pairs()
+    
+    L_cofs = []
+    for dim in dimensions:
+
+        try:
+            cof = cof_pp[0][dim]
+        except IndexError:
+            cof = np.array([])
+
+        L_cofs.append(np.array(cof, dtype=np.int32))
+
+    return L_cofs
+
+class CubicalLayer(tf.keras.layers.Layer):
+    """
+    TensorFlow layer for computing the persistent homology of a cubical complex
+    """
+    def __init__(self, dimensions, min_persistence=None, **kwargs):
+        """
+        Constructor for the CubicalLayer class
+
+        Parameters:
+            dimensions (List[int]): homology dimensions
```

Comment:
I now notice that `CubicalComplex.__init__` also has a parameter called `dimensions` but with a completely different meaning (the shape of the array). Should we use something more explicit like `homology_dimensions`, or do we consider that it is ok with just `dimensions` and hopefully people won't get too confused?

Reply:
That makes sense. I am a bit more in favor of `homology_dimensions` though. If @VincentRouvreau is OK with it, I'll update the code.

Reply:
`homology_dimensions` works for me.
In the sklearn interface, @VincentRouvreau also accepts an int when we want to specify a single dimension, so that would read `homology_dimensions=2`, I hope it is clear enough that we are not asking for 2 unspecified homology groups in this case but we only want homology in dimension 2 :crossed_fingers:

Reply:
We can by even more explicit with `homology_dimension_list`, but maybe this is too much?

Reply:
@MathieuCarriere `homology_dimension_list` may be too restrictive as my [interface](https://github.com/VincentRouvreau/gudhi-devel/blob/8d7244b510eee0d7927c521117198ef286028acf/src/python/gudhi/sklearn/cubical_persistence.py#L50-L52) can accept an `int or a list of int`. When only one dimension is required, it is quite useful and simplifies the code.

Reply:
OK I'll stick with `homology_dimensions` then